hivemind/optim/decentralized_optimizers.py

```diff
@@ -49,19 +65,60 @@ def __init__(self, opt: torch.optim.Optimizer, dht: DHT, *, prefix: str, target_
                                          target_group_size=target_group_size, **kwargs)
         self.lock_parameters, self.update_event, self.stop_event = Lock(), Event(), Event()
 
+        if scheduler_cls:
+            self.scheduler = scheduler_cls(opt)
+
+        self.local_epoch = 0
+        self.report_progress_expiration = report_progress_expiration
+        self.max_allowed_epoch_difference = max_allowed_epoch_difference
+        self.total_steps_in_epoch = total_steps_in_epoch
+        self.report_progress_key = f"{prefix}.progress"
+        self.report_progress_event, self.fetch_decentralized_state_event = Event(), Event()
+        self.lock_scheduler_params = Lock()
+        self.decentralized_state = DecentralizedState(max_epoch=0, total_steps=0)
+        self.fetch_decentralized_state_event.set()
+        self._fetch_decentralized_state(initial=True)
```

Comment:
It may be more intuitive if you refactor this as such:
- `_fetch_decentralized_state` fetches it once and returns state
- `_fetch_decentralized_state_periodically` runs prev. function in a loop, awaiting fetch event for each iteration
- in `__init__`, you can just set `self.decentralized_state = self._fetch_decentralized_state`